src/engraving/dom/segment.cpp

```diff
@@ -546,7 +546,11 @@ void Segment::removeStaff(staff_idx_t staff)
 //---------------------------------------------------------
 
 void Segment::checkElement(EngravingItem* el, track_idx_t track)
-{
+{   
+    // prevent segmentation fault on out of bounds index
+    if (track >= m_elist.size()) {
```

Comment:
Since this is an unexpected condition, I think it would be good to use an assertion in this case. We have a macro for that:
```suggestion
    IF_ASSERT_FAILED(track < m_elist.size()) {
```
(The `IF_ASSERT_FAILED` macro is equivalent with `if (!…)`, i.e. note the negation.)

For release builds, this would have the same result as your current change, but in debug builds it would cause an assertion failure, so that we know that there is still something to fix.